src/app/ngrx/reducers/selectors/recent-episode.selectors.ts

```diff
@@ -13,3 +18,40 @@ export const selectRecentEpisode = createSelector(selectRecentEpisodeEntities, s
 export const selectRecentEpisodeLoaded = createSelector(selectRecentEpisodeState, getRecentEpisodeLoaded);
 export const selectRecentEpisodeLoading = createSelector(selectRecentEpisodeState, getRecentEpisodeLoading);
 export const selectRecentEpisodeError = createSelector(selectRecentEpisodeState, getRecentEpisodeError);
+
+export const selectRecentEpisodeMetrics = createSelector(
+  selectRecentEpisode, selectEpisodeMetrics, (episode, metrics) => {
+    return metrics.find(m => m.id === episode.id);
+  }
+);
+
+export const selectRecentEpisodePerformanceMetrics = createSelector(
+  selectRecentEpisode, selectEpisodePerformanceMetricsEntities, selectRouter, selectRecentEpisodeMetrics,
+  (episode: EpisodeModel, entities, routerState, metrics): EpisodePerformanceMetricsModel => {
+    if (episode) {
+      if (metrics) {
+        // this whole bit can't reconcile totals if the app was loaded from non default route without recent episode metrics data
+        // if we have full metrics data on the recent episode, check the today and all time total values and reconcile
+        let totalForToday = entities[episode.id].today;
+        // if recent episode is not on routed episode page, check for episode dailyDownloads and total
+        const totalForRange = getTotal(metricsData(routerState, metrics)) || (metrics.dailyDownloads && getTotal(metrics.dailyDownloads));
+
+        // today's total doesn't update until day after release, so attempt to get it from the daily downloads
+        // (If app was loaded from non default url, we probably won't have dailyDownloads for today, an edge case I can live with.
+        // _could_ also total hourlies from today if they are refreshing the page from that route, but this is not doing that.)
+        if (metrics.dailyDownloads &&
+          new Date(metrics.dailyDownloads[metrics.dailyDownloads.length - 1][0]).valueOf() >= beginningOfTodayUTC().valueOf()) {
```

Comment:
Do you need to check the date for this?  Or could you just always pick the larger number (and default to 0).  Date comparisons just always make me nervous.  And who knows - maybe Castle will get fixed someday?

Reply:
Or I could just be misreading this.

Reply:
yeah, I it does need to check the date because if you loaded the app from a non default route that doesn't include today's date but does include the most recent episode, the last data point wouldn't be the today value. We probably will want to fix Castle eventually as I've heard that day 0, 7 days, and 30 days (but by 30 we really mean 28) are the important rollups for episodes, so we'll likely want those on a dashboard.

Reply:
(they could load the app on a non default route if say for example they refreshed a page with yesterday's url where the end date is no longer the current day)

Reply:
oh, right.  makes good sense then!